src/models/CommonModel.ts

```diff
@@ -188,73 +296,16 @@ export class CommonModel extends CommonSchema<CommonModel> {
     }
 
     // Which values are correct to use here? Is allOf required?
-    if (mergeFrom.$id !== undefined) {
+    if (mergeFrom.$id !== undefined && mergeTo.$id === undefined) {
       mergeTo.$id = mergeFrom.$id;
     }
-    if (mergeFrom.$ref !== undefined) {
+    if (mergeFrom.$ref !== undefined && mergeTo.$ref === undefined) {
       mergeTo.$ref = mergeFrom.$ref;
     }
-    if (mergeFrom.extend !== undefined) {
+    if (mergeFrom.extend !== undefined && mergeTo.extend === undefined) {
       mergeTo.extend = mergeFrom.extend;
```

Comment:
You can make here trick like:

```ts
mergeTo.$id = mergeTo.$id || mergeFrom.$id;
```

Also by this you don't need to write ifs

Reply:
Good call, looks cleaner :)